File: classes/parte-5-funciones-y-modularidad/083-cierres-closures-y-captura-de-variables/implementaciones/php/main.php
<?php
function hacerSumador($base) {
    return fn($x) => $base + $x;
}

$base = (int) trim(fgets(STDIN));
$sumar = hacerSumador($base);
echo "r1=" . $sumar(1) . " r2=" . $sumar(2) . "\n";

File: classes/parte-5-funciones-y-modularidad/083-cierres-closures-y-captura-de-variables/implementaciones/go/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func hacerSumador(base int) func(int) int {
	return func(x int) int {
		return base + x
	}
}

func main() {
	line, _ := bufio.NewReader(os.Stdin).ReadString('\n')
	base, _ := strconv.Atoi(strings.TrimSpace(line))
	sumar := hacerSumador(base)
	fmt.Printf("r1=%d r2=%d\n", sumar(1), sumar(2))
}
